test/libsolidity/semanticTests/array/pop/byte_array_pop_long_storage_empty_garbage_ref.sol

```diff
@@ -15,7 +15,7 @@ contract c {
 }
 // ----
 // test() ->
-// gas irOptimized: 142639
+// gas irOptimized: 142475
```

Comment:
I don't see any reverting paths here. Looking at IR, it's apparently because of the panic functions containing reverts but then why are so few cases affected? I'd expect these to be pretty much everywhere.

Reply:
Funny thing, actually :-).
After relaxing the inlining heuristics, in a lot of cases inlining of revert- and panic-helpers kicks in - after which this PR does not help anymore, since the canonical user-defined functions that always revert that occur most often are gone.

However, in this case here, we have an assembly block that writes to memory and is not marked ``memory-safe``, so we do not perform aggressive inlining, the panic helper stays uninlined and the PR kicks in.